### PR TITLE
Remove netcoreapp3.1

### PR DESCRIPTION
**Pull request by jennyf19**
- Add net7.0
- Structure refactoring to tests folder & devApps folder
- Add Directory.Build.props to tests
Have not tested all the dev apps, but have ensured they all build

### CODE REVIEW

**Comment by jennyf19** [comment]
https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1033276&view=results

File: Directory.Build.props [hunk 1]
```diff
@@ -85,9 +98,11 @@
     <MicrosoftExtensionsLoggingVersion>6.*</MicrosoftExtensionsLoggingVersion>
     <SystemTextEncodingsWebVersion>6.*</SystemTextEncodingsWebVersion>
     <MicrosoftExtensionsConfigurationBinderVersion>6.*</MicrosoftExtensionsConfigurationBinderVersion>
+    <MicrosoftExtensionsDependencyInjectionVersion>6.*</MicrosoftExtensionsDependencyInjectionVersion>
+
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(TargetFramework)' == 'net472' Or '$(TargetFramework)' == 'net462' Or '$(TargetFramework)' == 'net5.0'">
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net472' Or '$(TargetFramework)' == 'net462'">
```
Comment (jmprieur) [inline Directory.Build.props:105]: I believe that 472 could be a 2.x like netstandard2.0.
Let's check what George's doc or R9 do

File: tools/ConfigureGeneratedApplications/ConfigureGeneratedApplications - Backup.csproj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (jmprieur) [inline tools/ConfigureGeneratedApplications/ConfigureGeneratedApplications - Backup.csproj:1]: Do we need this file?

**Review by jmprieur — APPROVED** [review]
LGTM
Amazing work @jennyf19 
thank you!

File: tools/ConfigureGeneratedApplications/ConfigureGeneratedApplications - Backup.csproj (hunk 2 above)
Comment (jennyf19) [reply tools/ConfigureGeneratedApplications/ConfigureGeneratedApplications - Backup.csproj]: removed.

**Pull request merged**